mvnd native executable is not passing -Dkey=val to the daemon

I do have the following configuration for my `docker-maven-plugin` in the parent `pom.xml` for my multi module project:

```
	<plugin>
		<groupId>com.spotify</groupId>
		<artifactId>dockerfile-maven-plugin</artifactId>
		<version>1.4.13</version>
		<executions>
			<execution>
				<id>default</id>
				<goals>
					<goal>build</goal>
				</goals>
			</execution>
		</executions>
		<configuration>
			<repository>${docker.user}/${project.artifactId}</repository>
			<tag>${project.version}</tag>
			<buildArgs>
				<JAR_FILE>${project.build.finalName}.jar</JAR_FILE>
			</buildArgs>
		</configuration>
	</plugin>
```

each of my modules import this configuration using the following directive:

```
	<profiles>
		<profile>
			<id>docker</id>
			<activation>
				<property>
					<name>docker</name>
					<value>build</value>
				</property>
			</activation>
			<build>
				<plugins>
					<plugin>
						<groupId>com.spotify</groupId>
						<artifactId>dockerfile-maven-plugin</artifactId>
					</plugin>
				</plugins>
			</build>
		</profile>
	</profiles>	
```

With the command `mvn clean package -DskipTests -P docker -Ddocker.user=thiagolocatelli` project builds successfully and my docker images are created, however when I use the command `mvnd clean package -DskipTests -P docker -Ddocker.user=thiagolocatelli` the project fail with the following error messages:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for homeflix-parent 1.0.0-SNAPSHOT:
[INFO]
[INFO] homeflix-parent .................................... SUCCESS [  0.182 s]
[INFO] user-service ....................................... FAILURE [ 10.839 s]
[INFO] rental-service ..................................... FAILURE [ 10.839 s]
[INFO] cart-service ....................................... FAILURE [ 10.840 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.523 s (Wall Clock)
[INFO] Finished at: 2020-10-26T10:20:39-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.spotify:dockerfile-maven-plugin:1.4.13:build (default) on project user-service: Repo name "${docker.user}/user-service" must contain only lowercase, numbers, '-', '_' or '.'. -> [Help 1]
[ERROR] Failed to execute goal com.spotify:dockerfile-maven-plugin:1.4.13:build (default) on project rental-service: Repo name "${docker.user}/rental-service" must contain only lowercase, numbers, '-', '_' or '.'. -> [Help 1]
[ERROR] Failed to execute goal com.spotify:dockerfile-maven-plugin:1.4.13:build (default) on project cart-service: Repo name "${docker.user}/cart-service" must contain only lowercase, numbers, '-', '_' or '.'. -> [Help 1]
```

Not sure this is a bug related to `mvnd` but it is only happening with `mvnd` on my local environment. The same error happen with other projects as well.

--- a/client/pom.xml
+++ b/client/pom.xml
@@ -122,6 +122,7 @@
                                 -H:IncludeResources=org/jboss/fuse/mvnd/.*
                                 -H:IncludeResources=org/jline/utils/.*
                                 -H:IncludeResources=org/fusesource/jansi/jansi.properties
+                                -H:-ParseRuntimeOptions
                             </buildArgs>
                         </configuration>
                     </plugin>

--- a/client/src/main/java/org/jboss/fuse/mvnd/client/DefaultClient.java
+++ b/client/src/main/java/org/jboss/fuse/mvnd/client/DefaultClient.java
@@ -88,7 +88,6 @@ public class DefaultClient implements Client {
         boolean version = false;
         boolean showVersion = false;
         boolean debug = false;
-        final Properties commandLineProperties = new Properties();
         for (String arg : argv) {
             switch (arg) {
             case "-v":
@@ -111,9 +110,9 @@ public class DefaultClient implements Client {
                 if (arg.startsWith("-D")) {
                     final int eqPos = arg.indexOf('=');
                     if (eqPos >= 0) {
-                        commandLineProperties.setProperty(arg.substring(2, eqPos), arg.substring(eqPos + 1));
+                        System.setProperty(arg.substring(2, eqPos), arg.substring(eqPos + 1));
                     } else {
-                        commandLineProperties.setProperty(arg.substring(2), "");
+                        System.setProperty(arg.substring(2), "");
                     }
                 }
                 args.add(arg);

--- a/integration-tests/src/test/java/org/jboss/fuse/mvnd/it/ModuleAndPluginNativeIT.java
+++ b/integration-tests/src/test/java/org/jboss/fuse/mvnd/it/ModuleAndPluginNativeIT.java
@@ -42,11 +42,14 @@ public class ModuleAndPluginNativeIT {
     @Test
     void cleanInstall() throws IOException, InterruptedException {
         final Path helloPath = layout.multiModuleProjectDirectory().resolve("hello/target/hello.txt");
-        try {
-            Files.deleteIfExists(helloPath);
-        } catch (IOException e) {
-            throw new RuntimeException("Could not delete " + helloPath);
-        }
+        final Path helloPropertyPath = layout.multiModuleProjectDirectory().resolve("hello/target/hello.property.txt");
+        Stream.of(helloPath, helloPropertyPath).forEach(p -> {
+            try {
+                Files.deleteIfExists(p);
+            } catch (IOException e) {
+                throw new RuntimeException("Could not delete " + p);
+            }
+        });
 
         final Path localMavenRepo = layout.getLocalMavenRepository();
         TestUtils.deleteDir(localMavenRepo);
@@ -59,10 +62,13 @@ public class ModuleAndPluginNativeIT {
         /* Build #1: with "Hello" output to target/hello.txt */
         {
             final ClientOutput output = Mockito.mock(ClientOutput.class);
-            client.execute(output, "clean", "install", "-e", "-Dmvnd.log.level=DEBUG").assertSuccess();
+            client.execute(output, "clean", "install", "-e", "-Dmvnd.log.level=DEBUG", "-Dhello.property=Hello1")
+                    .assertSuccess();
 
             Assertions.assertThat(helloPath).exists();
             Assertions.assertThat(helloPath).usingCharset(StandardCharsets.UTF_8).hasContent("Hello");
+            Assertions.assertThat(helloPropertyPath).exists();
+            Assertions.assertThat(helloPropertyPath).usingCharset(StandardCharsets.UTF_8).hasContent("Hello1");
             Stream.of(installedJars).forEach(jar -> Assertions.assertThat(jar).exists());
         }
 
@@ -75,10 +81,12 @@ public class ModuleAndPluginNativeIT {
             final ClientOutput output = Mockito.mock(ClientOutput.class);
             client.execute(output,
                     "clean",
-                    "install", "-e", "-Dmvnd.log.level=DEBUG").assertSuccess();
+                    "install", "-e", "-Dmvnd.log.level=DEBUG", "-Dhello.property=Hello2").assertSuccess();
 
             Assertions.assertThat(helloPath).exists();
             Assertions.assertThat(helloPath).usingCharset(StandardCharsets.UTF_8).hasContent("Hi");
+            Assertions.assertThat(helloPropertyPath).exists();
+            Assertions.assertThat(helloPropertyPath).usingCharset(StandardCharsets.UTF_8).hasContent("Hello2");
             Stream.of(installedJars).forEach(jar -> Assertions.assertThat(jar).exists());
         }
 

--- a/integration-tests/src/test/projects/module-and-plugin/plugin/src/main/java/org/jboss/fuse/mvnd/test/module/plugin/mojo/HelloMojo.java
+++ b/integration-tests/src/test/projects/module-and-plugin/plugin/src/main/java/org/jboss/fuse/mvnd/test/module/plugin/mojo/HelloMojo.java
@@ -35,6 +35,9 @@ public class HelloMojo extends AbstractMojo {
     @Parameter
     File file;
 
+    @Parameter(property = "hello.property", defaultValue = "Hello")
+    String property;
+
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
 
@@ -42,6 +45,7 @@ public class HelloMojo extends AbstractMojo {
             final Path path = file.toPath();
             Files.createDirectories(path.getParent());
             Files.write(path, "Hello".getBytes(StandardCharsets.UTF_8));
+            Files.write(path.getParent().resolve("hello.property.txt"), property.getBytes(StandardCharsets.UTF_8));
         } catch (IOException e) {
             throw new RuntimeException("", e);
         }

